### DISCUSSION TITLE
Bosses not targeting NPCBots with Player-only spells [AzerothCore]

### DISCUSSION THREAD

**Discussion by Dreathean** [trickerer/Trinity-Bots#564] [General]
Hello, I looked and saw in a previous Issue that you had tried something to make Bosses target NPCBots with player-only spells, but that it caused issues including crashes. I imagine that is still the case to this day?

Do you know if that is something we would be able to adjust on our end on a spell-by-spell basis? It is really noticeable on a lot of bosses. But if not, that is not a big deal - thank you for this wonderful addition, it has breathed a lot of life and adds so much to our experience.

**Comment by trickerer** [comment]
There are several core layers where player-only checks exit and they need to be traversed for each spell you would want bosses to cast on bots if they are (effectively) player-only.
A spell is only cast on players if it either:
1) Has PLAYERS_ONLY spell attribute (AttibutesEx3, actual name is different between cores)
2) Has its targeting handled by UnitAI::SelectTarget() which has playersOnly parameter set to true
3) Has its targeting handled by custom target selector inside creature's script
4) Has its targeting handled by smart script with player selection set explicitly
5) Has an effect handler script / triggers a spell which is player-only (again, effectively, see 1), 4))

This is why it's a problem almost every time, especially if you want to allow selecting bots specifically and not pets and stuff. There is also a conditions thing possibly but i'm not sure.

So far I only work with spells like this if there is a public demand. Keep in mind that I don't play the game almost at all, since 2015 or so.

**Reply by Dreathean** [reply]
Got it, that makes sense. Thank you for the explanation, it definitely seems like a challenging and cumbersome issue. I hope you'll find enjoyment with the game again sometime!
